crates/solana-gateway-task-processor/src/component.rs

```diff
@@ -445,13 +454,17 @@ async fn incoming_message_already_executed(
     solana_rpc_client: &RpcClient,
     incoming_message_pda: &Pubkey,
 ) -> eyre::Result<bool> {
-    let raw_incoming_message = solana_rpc_client
+    if let Ok(raw_incoming_message) = solana_rpc_client
         .get_account_data(incoming_message_pda)
-        .await?;
-    let incoming_message = IncomingMessage::read(&raw_incoming_message)
-        .ok_or_eyre("failed to read incoming message")?;
+        .await
+    {
+        let incoming_message = IncomingMessage::read(&raw_incoming_message)
+            .ok_or_eyre("failed to read incoming message")?;
 
-    Ok(incoming_message.status.is_executed())
+        Ok(incoming_message.status.is_executed())
+    } else {
+        Ok(false)
+    }
```

Comment:
I was having issues here, but didn't look into them yet, my change was basically to ignore the error.